### PR TITLE
Rework the music resume logic, replace std::lock_guard by std::scoped_lock

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
I believe that this is much more stable (and also a bit simpler I think) implementation, current implementation in `master` branch is  deadlock-prone because of acquiring audio mutexes in `SDL_Mixer` callbacks. Implementation from this PR doesn't acquire any mutexes in callbacks and, as far as I can tell, works well and is much more suitable for the new release - we don't want our users to suffer. But it still needs to be extensively tested, if possible.

Also I heeded to persistent recommendations from SonarCloud and replaced `std::lock_guard` by `std::scoped_lock` from C++17. Honestly, if there is no need to safely acquire multiple locks at the same time, I'd prefer old `std::lock_guard`, because due to the variadic nature of `scoped_lock`'s constructor you can easily make a mistake and write like this:

```cpp
std::scoped_lock lock;
```

which is no-op. `lock_guard` will not allow this.

### CODE REVIEW

File: src/engine/audio.cpp [hunk 1]
```diff
@@ -551,6 +450,9 @@ void Audio::Quit()
     Music::Stop();
     Mixer::Stop();
 
+    Mix_ChannelFinished( nullptr );
+    Mix_HookMusicFinished( nullptr );
```
Comment (ihhub) [inline src/engine/audio.cpp:454]: We do this call within `Music::Stop()` function. I think we don't need it.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , I left one comment here. Could you please take a look at it?

File: src/engine/audio.cpp (hunk 1 above)
Comment (oleg-derevenetz) [reply src/engine/audio.cpp:454]: Ah, yes, really. I'll remove it.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@oleg-derevenetz , great work! Thank you!

**Pull request merged**